shogun-boot/src/main/java/de/terrestris/shogun/boot/config/KeycloakBootWebSecurityConfig.java

```diff
@@ -0,0 +1,66 @@
+/* SHOGun, https://terrestris.github.io/shogun/
+ *
+ * Copyright © 2020-present terrestris GmbH & Co. KG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.terrestris.shogun.boot.config;
+
+import de.terrestris.shogun.config.DefaultWebSecurityConfig;
+import org.keycloak.adapters.springsecurity.KeycloakConfiguration;
+import org.keycloak.adapters.springsecurity.authentication.KeycloakAuthenticationProvider;
+import org.keycloak.adapters.springsecurity.config.KeycloakWebSecurityConfigurerAdapter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.core.authority.mapping.SimpleAuthorityMapper;
+import org.springframework.security.core.session.SessionRegistryImpl;
+import org.springframework.security.web.authentication.session.RegisterSessionAuthenticationStrategy;
+import org.springframework.security.web.authentication.session.SessionAuthenticationStrategy;
+
+@ConditionalOnExpression("${keycloak.enabled:true}")
+@Configuration
+@EnableWebSecurity
+@KeycloakConfiguration
+public class KeycloakBootWebSecurityConfig extends KeycloakWebSecurityConfigurerAdapter implements DefaultWebSecurityConfig {
```

Comment:
Did you see that the keycloak spring adapter will be deprecated soon? See

- https://www.keycloak.org/2022/02/adapter-deprecation
- https://www.keycloak.org/docs/latest/release_notes/index.html#spring-security-and-boot-adapter-deprecation
- https://www.springcloud.io/post/2022-02/keycloak-announces/#gsc.tab=0
- https://www.baeldung.com/spring-security-openid-connect

Just a general remark, and probably something for another MR. But maybe this will make the rework easier? Or more complicated? I'm not entirely sure about the impact on this...

Reply:
Good point, haven't noticed that during refactoring. Actually I'm not sure if this would become a trivial update, so I suggest to handle it in an upcoming PR, but I'll check it anyway.

Reply:
:+1: